src/main/webapp/components/TopNavigationMenu.js

```diff
@@ -0,0 +1,294 @@
+    Storage.setUserIsLogged(true);
+    window.location.reload();
+    this.init().then(resp => {
+      if (Storage.getLocationFrom() != null) {
+        this.props.history.push(Storage.getLocationFrom());
+        Storage.removeLocationFrom();
+      } else {
+        this.props.history.push("/index");
+      }
+    });
+  };
+
+  signOut = async () => {
+    this.props.showSpinner();
+    Storage.clearStorage();
+    if (window.gapi.auth2 != undefined) {
+      let auth2 = window.gapi.auth2.getAuthInstance();
+      await auth2.signOut();
+      await User.signOut();
+    }
+    this.setState({
+      isLogged: false
+    }, () => {
+      window.location.reload();
+    });
+  };
+
+  openMetricsReport() {
+    Reports.getMetricsReport();
+  }
+
+  loadOptions = (query, callback) => {
+    let options = [];
+    if (query.length > 2) {
+      Search.getMatchingIssues(query).then(response => {
+        let options = response.data.map((item) => {
+          return {
+            key: item.id,
+            value: item.value,
+            label: this.generateLabel(item),
+            linkDisabled: item.linkDisabled,
+            url: item.url,
+            isDisabled: item.linkDisabled
+          };
+        });
+        callback(options);
+      }).catch(e => {
+        callback(options)
+      });
+    } else {
+      callback(options)
+    }
+  };
+
+  generateLabel = (item) => {
+    let label = item.label;
+    if (item.linkDisabled === true && item.pm.length > 0) {
+      label = this.createLabel(item.pm, item.label);
+    } else if (item.linkDisabled === true) {
+      label = this.createLabel(item.reporter, item.label);
+    }
+    return label;
+  }
+
+  createLabel (contact, label) {
+    return h(span, { style: styles.listResultContainer }, [
+      span({ style: styles.badgeContactAccess }, [
+        ' Please contact ' + contact + ' for access'
+      ]),
+      div({ style: { 'marginTop': '10px' } }, [
+        label
+      ])
+    ])
+  }
+
+  handleSearchChange = (data, action) => {
+    if (data != null && !data.linkDisabled) {
+      window.location.href = data.url;
+    }
+  };
+
+  render() {
+    return (
+      div({ className: "navbar navbar-default navbar-fixed-top", role: "navigation" }, [
+        div({ className: "container" }, [
+          div({ className: "navbar-header" }, [
+            button({ className: "navbar-toggle", type: "button" }, [
+              span({ className: "sr-only" }, ["Toggle navigation"]),
+              span({ className: "icon-bar" }, []),
+              span({ className: "icon-bar" }, []),
+              span({ className: "icon-bar" }, [])
+            ]),
+            h(Link, { className: "navbar-brand", to: { pathname: Storage.userIsLogged() ? UrlConstants.index : '/' } },
+              [
+                span({}, [
+                  "ORSP Portal ",
+                  span({ isRendered: process.env.NODE_ENV === 'development', className: "label label-danger" }, ["Dev"])
```

Comment:
Note that this still seems like an unresolved question. Where is `process.env.NODE_ENV` being set at the environment level? We do use it in at least one other place, but I don't see where we are setting this value.

Reply:
Hi Greg. I think this link has the information that we need: 
https://create-react-app.dev/docs/adding-custom-environment-variables

The environment variables are embedded during the build time. These environment variables will be defined on process.env. There is also a special built-in environment variable called NODE_ENV. You can read it from process.env.NODE_ENV. When you run npm start, it is always equal to 'development', when you run npm test it is always equal to 'test', and when you run npm run build to make a production bundle, it is always equal to 'production'. These environment variables can be useful for displaying information conditionally based on where the project is deployed or consuming sensitive data that lives outside of version control.

Reply:
OK, that helps explain where it *could* be used. The application doesn't run under npm, it's a native java app. I'm pretty sure this doesn't exist in our current build environment, so this feature doesn't look like it will work. I'm also fine with moving this to a separate story since the implementation is pretty unrelated to the goal of the PR. In that story, we should address where the other use of it is as well - that is likely broken.

Reply:
Greg. I removed process.env.NODE_ENV. I'm getting this value from another place.

<%@ page import="grails.util.Environment" %>
 env: "${Environment.current.name}"

Reply:
Looks good! 👍